service/service_test.go

```diff
@@ -123,21 +122,24 @@ func TestService_Recovery(t *testing.T) {
 	s.openRoundMutex.Lock()
 	req.Nil(s.openRound)
 	s.openRoundMutex.Unlock()
+	s.Lock()
 	req.Equal(len(s.executingRounds), 0)
-
+	s.Unlock()
 	// Create a new service instance.
-	sig = signal.NewSignal()
-	s, err = NewService(sig, cfg, tempdir)
+	s, err = NewService(cfg, tempdir)
 	req.NoError(err)
 
 	err = s.Start(broadcaster)
 	req.NoError(err)
 
 	// Service instance should recover 2 rounds: round 1 in executing state, and round 2 in open state.
-	req.Equal(1, len(s.executingRounds))
+	req.Eventually(func() bool { s.Lock(); defer s.Unlock(); return len(s.executingRounds) == 1 }, time.Second, time.Millisecond)
```

Comment:
Does this fix one of our flaky tests? If yes good find 👍 

I would just increase the check interval from `time.Millisecond` to maybe `20 * time.Millisecond` to give more time to the service where the mutex isn't locked.

Reply:
It's better but still can be flaky because the round expected to be executing might already finish. A millisecond is still a long time, increasing it will make it more flaky :(

IMO the tests generally shouldn't inspect the inner (private) state of Service which is extremely asynchronous (rounds start and finish asynchronously) but rather make assertions at the observable API level. But that's a bigger tests refactor.